### PR TITLE
MAT-5745: Quantity Attribute Input

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-5745](https://jira.cms.gov/browse/MAT-5745)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/DisplayAttributeInputs.tsx [hunk 1]
```diff
@@ -88,6 +91,19 @@ const DisplayAttributeInputs = ({
...
         );
+
+      case "Quantity":
+        return (
+          <QuantityInput
+            quantity={{}}
+            onQuantityChange={(val) => {
+              if (val.unit && val.value) {
+                setAttributeValue(new CQL.Quantity(val.value, val.unit));
+              }
```
Comment (adongare) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/DisplayAttributeInputs.tsx]: Quantity object creation should happen in the Quantity component. No need to check here again.

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/DisplayAttributeInputs.tsx [hunk 2]
```diff
@@ -73,7 +74,9 @@ const DisplayAttributeInputs = ({
             data-testid="ratio-input"
             onRatioChange={(newCQLRatio) => {
               setCurrentRatio(newCQLRatio);
-              setAttributeValue(newCQLRatio);
+              if (newCQLRatio.numerator.unit && newCQLRatio.denominator.unit) {
```
Comment (adongare) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/DisplayAttributeInputs.tsx]: this check is unnecessary

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/AttributeSection.test.tsx [hunk 3]
```diff
@@ -485,4 +485,82 @@ describe("AttributeSection", () => {
     expect(typeOptions.length).toEqual(9);
     fireEvent.click(within(typeSelect).getByText("Decimal"));
   });
+
+  it("shows DateTime on selecting the DateTime type", async () => {
+    const assessmentElement: AssessmentPerformed = new AssessmentPerformed();
```
Comment (adongare) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/AttributeSection.test.tsx]: I think no need to provide explicit type here. `const assessmentElement = new AssessmentPerformed();` should be fine. Typescript can infer the type from the right side assignment.

File: src/components/common/ratioInput/RatioInput.tsx [hunk 4]
```diff
@@ -32,7 +32,7 @@ const RatioInput = ({
           <Quantity
             quantity={ratio.numerator}
             canEdit={canEdit}
-            label={label}
+            label={`Numerator`}
```
Comment (adongare) [inline src/components/common/ratioInput/RatioInput.tsx]: I think a regular string would be fine. template literal is not needed here.

File: src/components/common/ratioInput/RatioInput.tsx [hunk 5]
```diff
@@ -41,7 +41,7 @@ const RatioInput = ({
           <Quantity
             quantity={ratio.denominator}
             canEdit={canEdit}
-            label={label}
+            label={`Denominator`}
```
Comment (adongare) [inline src/components/common/ratioInput/RatioInput.tsx]: same here

File: src/components/common/ratioInput/RatioInput.test.tsx [hunk 6]
```diff
@@ -112,18 +132,26 @@ describe("RatioInput Component", () => {
...
-    expect(screen.getByText("Ratio")).toBeInTheDocument();
+    expect(screen.getByText("Numerator")).toBeInTheDocument();
+    expect(screen.getByText("Denominator")).toBeInTheDocument();
     expect(screen.getAllByPlaceholderText("value").length).toBe(2);
     expect(screen.getAllByPlaceholderText("unit").length).toBe(2);
 
-    const valueInput = screen.getAllByTestId(
-      "quantity-value-input-Ratio"
+    const valueInputNumerator = screen.getAllByTestId(
+      `quantity-value-input-Numerator`
```
Comment (adongare) [inline src/components/common/ratioInput/RatioInput.test.tsx]: not sure why the consistent lowercase style was not followed- `quantity-value-input-Numerator` vs `quantity-value-input-numerator`. Also template literal her as well

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/DisplayAttributeInputs.tsx (hunk 2 above)
Comment (ethankaplan) [reply src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/DisplayAttributeInputs.tsx]: image
Without it, this error happens

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/DisplayAttributeInputs.tsx (hunk 1 above)
Comment (ethankaplan) [reply src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/DisplayAttributeInputs.tsx]: Done

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/DisplayAttributeInputs.tsx (hunk 2 above)
Comment (adongare) [reply src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/DisplayAttributeInputs.tsx]: this check should happen in the Quantity component itself. I don’t think there will be quantity without a unit. also, onChange should not be triggered until the quantities and ratio is valid

File: src/components/common/ratioInput/RatioInput.test.tsx (hunk 6 above)
Comment (ethankaplan) [reply src/components/common/ratioInput/RatioInput.test.tsx]: Addressed

**Review by adongare — APPROVED** [review]

**Pull request merged**